### ISSUE TITLE
Remove global Codec variable

### ISSUE THREAD

**Issue by joshua-kim** [ava-labs/coreth#424]
The use of the global codec variable makes VMs difficult to test. We should have a way to supply a codec to the VM for testing purposes.

**Comment by ceyonur** [comment]
is this a specific problem to Coreth? I think this is what we currently follow in all VMs, and it should probably be changed from AvalancheGo.

**Comment by darioush** [comment]
I don't think this is a goal we are actively pursuing, we can re-open an issue if needed.
